File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/Alternative.java
package quintype.com.templatecollectionwithrx.models.story;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Alternative implements Parcelable {

    @SerializedName("home")
    @Expose
    private Home home;

    public static final Creator<Alternative> CREATOR = new Creator<Alternative>() {
        @Override
        public Alternative createFromParcel(Parcel in) {
            return new Alternative(in);
        }

        @Override
        public Alternative[] newArray(int size) {
            return new Alternative[size];
        }
    };

    /**
     * @return The home
     */
    public Home getHome() {
        return home;
    }

    /**
     * @param home The home
     */
    public void setHome(Home home) {
        this.home = home;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(this.home, flags);
    }

    public Alternative() {
    }

    protected Alternative(Parcel in) {
        this.home = in.readParcelable(Home.class.getClassLoader());
    }

}

File: androidcore/src/main/java/com/example/androidcore/models/story/Alternative.kt
package com.example.androidcore.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */
class Alternative : Parcelable {

    /**
     * @return The home
     */
    /**
     * @param home The home
     */
    @SerializedName("home")
    @Expose
    var home: Home? = null


    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeParcelable(this.home, flags)
    }

    protected constructor(`in`: Parcel) {
        this.home = `in`.readParcelable(Home::class.java.classLoader)
    }

    companion object CREATOR : Parcelable.Creator<Alternative> {
        override fun createFromParcel(parcel: Parcel): Alternative {
            return Alternative(parcel)
        }

        override fun newArray(size: Int): Array<Alternative?> {
            return arrayOfNulls(size)
        }
    }

}